### PR TITLE
Lets get the circleci build generating images too. And display the yarn cache dir being used during the build.  fix: s2i-image name was not correct fix: copy in the locally built SNAPSHOT artifacts into the docker build

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: s2i/pom.xml [hunk 1]
```diff
@@ -80,21 +80,36 @@
...
+            </configuration>
+          </execution>
+          <execution>
+            <id>copy-local-repo</id>
+            <phase>package</phase>
+            <goals>
+              <goal>exec</goal>
+            </goals>
+            <configuration>
+              <executable>bash</executable>
```
Comment (rhuss) [inline s2i/pom.xml]: won't work on windows obviously. Maybe we should add this to the README.

File: s2i/pom.xml [hunk 2]
```diff
@@ -80,21 +80,36 @@
...
+              <arguments>
+                <argument>${basedir}/copy_mvn_repo.sh </argument>
+                <argument>${syndesis.rest.version}</argument>
```
Comment (rhuss) [inline s2i/pom.xml]: Why is there a specific "rest" version ? Should be always 'syndesis.version' (one of the benefits of a monorepo)

File: s2i/copy_mvn_repo.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      <id>local</id>
+      <url>file:/deployments/project/.m2/repository</url>
+    </snapshotRepository>
+    <repository>
+      <id>local</id>
+      <url>file:/deployments/project/.m2/repository</url>
+    </repository>
+  </distributionManagement>
+"""
+perl -p -i -e "s|</project>|${XML}</project>|" "$TARGET_DIR/pom.xml"
```
Comment (rhuss) [inline s2i/copy_mvn_repo.sh]: Love it ;-) But think sed could be good  enough here, too (so that it works for systems without perl, too).

Comment (rhuss) [reply s2i/copy_mvn_repo.sh]: But I'm fine with this of course (actually I would rewrite it into a complete perl script or maybe even groovy script to make it cross plattform)

File: s2i/pom.xml [hunk 4]
```diff
@@ -80,21 +80,36 @@
...
+                <argument>${basedir}/src/main/docker/project</argument>
```
Comment (rhuss) [inline s2i/pom.xml]: Copying into a directory below `src/` is not so nice

File: s2i/copy_mvn_repo.sh (hunk 3 above)
Comment (rhuss) [reply s2i/copy_mvn_repo.sh]: However, the script has a serious flaw: You will add endlessly `<distributionManagement>` sections to `src/main/docker/pom.xml` regardless of whether this section is already included or not. Also modifying committed source files has issues (remember the connectors ?)

File: build.sh [hunk 5]
```diff
@@ -74,14 +74,19 @@ function init_options() {
...
-  # If  we are running in cicleci lets configure thigs to avoid running out of memory:
+  # If  we are running in cicleci 
   if [ "$CIRCLECI" == "true" ] ; then
-    MAVEN_PARAMS="$MAVEN_PARAMS -Dbasepom.test.fork-count=2"
+    # Use batch mode so we get friendiler log files.
+    MAVEN_PARAMS="$MAVEN_PARAMS --batch-mode"
+    # lets configure thigs to avoid running out of memory:
+    MAVEN_PARAMS="$MAVEN_PARAMS -Dbasepom.test.fork-count=2 --batch-mode -P image"
+    # Enable the image profile so that they get built
+    MAVEN_PARAMS="$MAVEN_PARAMS -P image"
```
Comment (rhuss) [inline build.sh]: Tomorrow I will update build.sh to be more flexible with respect to combining profiles from commandline arguments given to the script. I know that profiles dont work optimally in Maven (e.g. the default profile is only activated if no other profile is activated), but I hope I can alleviate the pain with nice script options.

File: s2i/copy_mvn_repo.sh (hunk 3 above)
Comment (chirino) [reply s2i/copy_mvn_repo.sh]: the pom.xml is generated on every mvn build, so no endless dups of '<distributionManagement>'.   Tried sed.  It fails to handle '\n''s well.  Our build.sh is based on bash already.  Seem we are already tied down to bash based systems.

File: s2i/pom.xml (hunk 1 above)
Comment (chirino) [reply s2i/pom.xml]: This would be just one item in a long list no?

File: s2i/pom.xml (hunk 2 above)
Comment (chirino) [reply s2i/pom.xml]: just reusing what was defined previously.

File: s2i/pom.xml (hunk 4 above)
Comment (chirino) [reply s2i/pom.xml]: Sure but that's where it's been currently generating to.  Moving to target would be better tho.

**Pull request merged**